File: lab_2/ex3.py
class C:
    def __init__(self):
        self.abc = self.A()
        self.xy = self.B()
    
    class A:
        def vvod1(self):
            self.a = int(input("a: "))
            self.b = int(input("b: "))
            self.c = int(input("c: "))
              
    class B:
        def vvod2(self, x, y):
            self.x = x
            self.y = y
            
    def sum(self):
        s = self.abc.a + self.abc.b + self.abc.c + self.xy.x + self.xy.y
        return s
    
    # def mult(self):
    #     m = self.abc.a * self.abc.b * self.abc.c * self.xy.x * self.xy.y
    #     return m

tmp = C()
# tmp1 = C()

tmp.A.vvod1(4, 2, 6,)
# tmp.B.vvod2(3, 6)

# tmp1.vvod1(1, 1 ,1)
# tmp1.vvod2(2, 9)

print("First:")
print("\tSum:", tmp.sum())
# print("\tMult:", tmp.mult())

# print("Second:")
# print("\tSum:", tmp1.sum())
# print("\tMult:", tmp1.mult())

File: lab_2/cpp/ex3.cpp
#include <iostream>

using namespace std;

class Third {
   public:
    class First {
       public:
        int a, b, c;
        void vvod1() {
            cout << "a = ";
            cin >> a;
            cout << "b = ";
            cin >> b;
            cout << "c = ";
            cin >> c;
        }
    };
    class Second {
       public:
        float x, y;
        void vvod2() {
            cout << "x = ";
            cin >> x;
            cout << "y = ";
            cin >> y;
        }
    };
    float s, m;
    void sum(First &first, Second &second) {
        s = first.a + first.b + first.c + second.x + second.y;
        cout << s << endl;
    };
    void mult(First &first, Second &second) {
        m = first.a * first.b * first.c * second.x * second.y;
        cout << m << endl;
    };
};

int main(int argc, char const *argv[]) {
    Third::First f;
    f.vvod1();

    Third::Second s;
    s.vvod2();

    Third tmp;
    tmp.sum(f, s);
    tmp.mult(f, s);
}
